### PR TITLE
*: clean up code for insert/update statement

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
1. metrics of batch get will always inc(1) even if nothing to get.
1. `kv.SkipCheckForWrite` is unused for a long while https://github.com/pingcap/tidb/pull/3600, make a cleanup.
1. some other cleanup.

### What is changed and how it works?
Fix them.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported variable/fields change

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8867)

### CODE REVIEW

**Comment by ngaut** [comment]
We should remove SkipCheckForWrite completely.

File: store/tikv/snapshot.go [hunk 1]
```diff
@@ -67,6 +67,10 @@ func (s *tikvSnapshot) SetPriority(priority int) {
 // BatchGet gets all the keys' value from kv-server and returns a map contains key/value pairs.
 // The map will not contain nonexistent keys.
 func (s *tikvSnapshot) BatchGet(keys []kv.Key) (map[string][]byte, error) {
+	m := make(map[string][]byte)
+	if len(keys) == 0 {
+		return m, nil
+	}
```
Comment (lysu) [inline store/tikv/snapshot.go]: prefer:

```
 	if len(keys) == 0 {
 		return make(map[string][]byte), nil
 	}
```

File: table/tables/tables.go [hunk 2]
```diff
@@ -447,12 +447,6 @@ func (t *tableCommon) AddRecord(ctx sessionctx.Context, r []types.Datum, skipHan
 	}
 
 	sessVars := ctx.GetSessionVars()
-	// when LightningMode or BatchCheck is true,
-	// no needs to check the key constrains, so we names the variable skipCheck.
-	skipCheck := sessVars.LightningMode || ctx.GetSessionVars().StmtCtx.BatchCheck
-	if skipCheck {
-		txn.SetOption(kv.SkipCheckForWrite, true)
-	}
```
Comment (lysu) [inline table/tables/tables.go]: so we should remove `kv.SkipCheckForWrite` too~?

File: table/tables/tables.go [hunk 3]
```diff
@@ -565,7 +559,7 @@ func (t *tableCommon) addIndices(ctx sessionctx.Context, recordID int64, r []typ
 			return 0, errors.Trace(err2)
 		}
 		var dupKeyErr error
-		if !skipCheck && (v.Meta().Unique || v.Meta().Primary) {
+		if !skipCheck && v.Meta().Unique {
```
[the remark this answers is not in the document]
Comment (jackysp) [reply table/tables/tables.go]: If `Primary` is true, `Unique` will be true.

**Pull request merged**